src/org.hdfgroup.object/hdf/object/h5/H5MetaDataContainer.java

```diff
@@ -0,0 +1,286 @@
+/*****************************************************************************
+ * Copyright by The HDF Group.                                               *
+ * Copyright by the Board of Trustees of the University of Illinois.         *
+ * All rights reserved.                                                      *
+ *                                                                           *
+ * This file is part of the HDF Java Products distribution.                  *
+ * The full copyright notice, including terms governing use, modification,   *
+ * and redistribution, is contained in the files COPYING and Copyright.html. *
+ * COPYING can be found at the root of the source code distribution tree.    *
+ * Or, see https://support.hdfgroup.org/products/licenses.html               *
+ * If you do not have access to either file, you may request a copy from     *
+ * help@hdfgroup.org.                                                        *
+ ****************************************************************************/
+
+package hdf.object.h5;
+
+import java.util.List;
+import java.util.Vector;
+
+import hdf.object.AttributeDataset;
+import hdf.object.FileFormat;
+import hdf.object.HObject;
+import hdf.object.MetaDataContainer;
+
+/**
+ * An class that provides general I/O operations for object metadata
+ * attached to an object. For example, reading metadata content from the file
+ * into memory or writing metadata content from memory into the file.
+ * <p>
+ *
+ * @see hdf.object.HObject
+ *
+ * @version 2.0 4/2/2018
+ * @author Peter X. Cao, Jordan T. Henderson
+ */
+@SuppressWarnings("rawtypes")
+public class H5MetaDataContainer extends HObject implements MetaDataContainer {
```

Comment:
MetaDataContainer is an interface. It doesn't make sense to have an H5MetaDataContainer unless it's an extension of a MetaDataContainer. A MetaDataContainer isn't supposed to be an actual object.

Reply:
I think here we can probably put all of the metadata operations into one of the base classes (probably Dataset and Attribute) and cover all the bases.